### PR TITLE
update 'exitAllPositions' scUsdcV2

### PR DESCRIPTION
**Pull request by brkomir**
- changed caller for exitAllPositions from admin to keeper
- removed "underwater" requirement so all positions can be exited anytime

### CODE REVIEW

File: src/steth/scUSDCv2.sol [hunk 1]
```diff
@@ -144,30 +138,35 @@ contract scUSDCv2 is BaseV2Vault {
...
-        amounts[0] = debt - wethBalance;
+            uint256 wethLeft = _wethBalance();
 
-        _initiateFlashLoan();
-        balancerVault.flashLoan(address(this), tokens, amounts, abi.encode(FlashLoanType.ExitAllPositions));
-        _finalizeFlashLoan();
+            if (wethLeft != 0) _swapWethForUsdc(wethLeft, 0);
+        }
 
         if (usdcBalance() < _endUsdcBalanceMin) revert EndUsdcBalanceTooLow();
```
[the remark this answers is not in the document]
Comment (brkomir) [reply src/steth/scUSDCv2.sol:171]: why do you think that? the expectation is that at the end  the amount of usdc we have in the vault is at least the amount provided as function param (float included)

[the remark this answers is not in the document]
Comment (dxganta) [reply src/steth/scUSDCv2.sol:171]: I see. Then you should add it to the comments too. It seemed like this is the usdc balance got after exiting the positions, i.e the total Invested amount.

[the remark this answers is not in the document]
Comment (brkomir) [reply src/steth/scUSDCv2.sol:171]: the `usdcBalance()` function returns usdc held by the vault, so I don't see how that could confuse you. Are you referring to the `_endUsdcBalanceMin` param?

[the remark this answers is not in the document]
Comment (dxganta) [reply src/steth/scUSDCv2.sol:171]: Yes the _endUsdcBalancemin() parameter.

[the remark this answers is not in the document]
Comment (brkomir) [reply src/steth/scUSDCv2.sol:173]: [fixed](https://github.com/lindy-labs/sandclock-contracts/pull/125/commits/b19346e873786da917e5f2b3e6a1384ab87abb4a)

**Review by dxganta — APPROVED** [review]

**Pull request merged**